src/Orleans.Runtime/Silo/Silo.cs

```diff
@@ -344,7 +339,7 @@ public async Task StopAsync(CancellationToken cancellationToken)
 
             try
             {
-                await this.lifecycleSchedulingSystemTarget.QueueTask(() => this.siloLifecycle.OnStop(cancellationToken)).ConfigureAwait(false);
+                await Task.Run(() => this.siloLifecycle.OnStop(cancellationToken), cancellationToken).ConfigureAwait(false);
```

Comment:
We should pass `CancellationToken.None` to `Task.Run` here, since we must call `OnStop` even if the provided `cancellationToken` is already canceled.